src/QueryM/WaitUntilSlot.purs

```diff
@@ -137,3 +142,39 @@ posixTimeToSeconds (POSIXTime futureTimeBigInt) = do
     $ map (wrap <<< Int.toNumber)
     $ BigInt.toInt
     $ futureTimeBigInt / BigInt.fromInt 1000
+
+-- | Wait at least `offset` number of slots.
+waitNSlots :: Natural -> QueryM Chain.Tip
+waitNSlots offset = do
+  offsetBigNum <- liftM (error "Unable to convert BigInt to BigNum")
+    $ (BigNum.fromBigInt <<< Natural.toBigInt) offset
+  if offsetBigNum == BigNum.fromInt 0 then getChainTip
+  else do
+    slot <- currentSlot
+    newSlot <- liftM (error "Unable to advance slot")
+      $ wrap <$> BigNum.add (unwrap slot) offsetBigNum
+    waitUntilSlot newSlot
+
+currentSlot :: QueryM Slot
+currentSlot = getChainTip >>= case _ of
+  Chain.Tip (Chain.ChainTip { slot }) -> pure slot
+  Chain.TipAtGenesis -> (pure <<< Slot <<< BigNum.fromInt) 0
```

Comment:
Given both cases simply `pure`, we can use `<#>` instead of `>>=`

Reply:
I didn't know of that operator, nice! thanks.